### PR TITLE
Add docs for nats onramp and offramp

### PR DESCRIPTION
**Pull request by aryan9600**
Signed-off-by: Sanskar Jaiswal <jaiswalsanskar078@gmail.com>

Docs for the NATS offramp and onramp. [Related PR](https://github.com/tremor-rs/tremor-runtime/pull/866)

### CODE REVIEW

File: docs/artefacts/offramps.md [hunk 1]
```diff
@@ -484,6 +485,57 @@ Compare And Swap operation. Those operations require old values to match what it
...
 
+### nats
+The nats offramp connects to nats server(s) and publishes a message to specified subject for every event.
+
+The default [codec](codecs.md#json) is `json`.
+
+Supported configuration operations are:
+
+- `hosts` - List of hosts to connect to.
+- `subject` - Subject for the to be published message.
```
Comment (mfelsche) [inline docs/artefacts/offramps.md:496]: `Subject for the message to be published` sounds a little better to me.

File: docs/artefacts/offramps.md [hunk 2]
```diff
@@ -484,6 +485,57 @@ Compare And Swap operation. Those operations require old values to match what it
...
+- `reply` - Optional string specifying the reply subject.
+- `headers` - Optional mapping specifying the headers for the message.
```
Comment (mfelsche) [inline docs/artefacts/offramps.md:498]: Are there any restrictions on what header values can be?

**Review by mfelsche — COMMENTED** [review]
Looks good, just some small nitpicks.

File: docs/artefacts/offramps.md (hunk 2 above)
Comment (aryan9600) [reply docs/artefacts/offramps.md:498]: yeah both key and values need to be a String
`Optional mapping (String: Vec<String>) specifying the headers for the message.`
does this work?

Comment (Licenser) [reply docs/artefacts/offramps.md:498]: I would avoid the rust specific sones (i.e. Vec) perhaps something like `Headers are key-value pairs with the key being a string and the value being either a string or a list of strings`?

**Review by mfelsche — APPROVED** [review]

**Comment by mfelsche** [comment]
Yay!

**Pull request merged**